testing/integration/upgrade_downgrade_test.go

```diff
@@ -61,33 +58,13 @@ func TestStandaloneDowngradeToSpecificSnapshotBuild(t *testing.T) {
 	// as the currently running binary (so, we don't have a file system collision).
 	// Multiple builds can have different IDs but the same commit hash.
 	preReleaseVersion := latestSnapshotVersion.VersionWithPrerelease()
-	resp, err := aac.GetBuildsForVersion(ctx, preReleaseVersion)
-	require.NoError(t, err)
-
-	if len(resp.Builds) < 2 {
-		t.Skipf("need at least 2 builds in the version %s", latestSnapshotVersion.VersionWithPrerelease())
-		return
-	}
-
-	t.Logf("found %d builds for version %q", len(resp.Builds), preReleaseVersion)
-
-	t.Logf("looking for a build that does not match the current commit hash %q", startVersion.Binary.Commit)
-	var upgradeVersionString string
-	for _, buildID := range resp.Builds[1:] {
-		details, err := aac.GetBuildDetails(ctx, preReleaseVersion, buildID)
-		require.NoError(t, err)
-		if details.Build.Projects[artifactElasticAgentProject].CommitHash != startVersion.Binary.Commit {
-			upgradeVersionString = buildID
-			break
-		}
-		t.Logf("build %q matches the current commit hash %q, skipping...", buildID, startVersion.Binary.Commit)
-	}
-
-	if upgradeVersionString == "" {
+	buildInfo, err := aac.FindBuild(ctx, preReleaseVersion, startVersion.Binary.Commit, 1)
```

Comment:
Why setting the offset to 1? Is it just to guarantee there will be at least 2 snapshots available? If so, is it necessary? There is already the build hash to guarantee the same build will not be used.
I'd understand using the offset to guarantee only newer builds are used, but from what understand, it's not the case

Reply:
This is already documented above, please read the comment.

Reply:
> If so, is it necessary? There is already the build hash to guarantee the same build will not be used.

I gave the same argument to @pchila (who wrote this test) in a call after his comment here https://github.com/elastic/elastic-agent/pull/4095#discussion_r1457430896 but apparently you and me don't understand the test well enough. So, I kept the logic as it was before.

Reply:
Adding this here, if the comment is not clear feel free to update that.

This test verifies that agent is able to upgrade to a *specific snapshot build* using `elastic-agent upgrade x.y.z-SNAPSHOT+<buildid>` . This functionality has been added on top of the existing "upgrade to latest snapshot build" which has the form of `elastic-agent upgrade x.y.z-SNAPSHOT`.

The reason why the test does not select the latest snapshot build is that the result would be indistinguishable between the 2 forms of the upgrade command (those would select the same build) so we could silently break the upgrade to specific snapshot feature while still having a green test.

Hope that helps...